### DISCUSSION TITLE
Why is image recognition not working for me? What am I missing?

### DISCUSSION THREAD

**Discussion by gnapse** [open-webui/open-webui#1652] [Q&A] [answered]
I tried to test the "upload" functionality by uploading an image to ask the LLM to describe it. It works from a UI point of view (it allows me to upload an image, and it shows it nicely in the chat UI) but the LLM does not seem to be aware of the image, as if it's not provided to it in the messages.

![CleanShot 2024-04-21 at 15 28 23@2x](https://github.com/open-webui/open-webui/assets/15199/f08ee849-423d-41cf-882d-8e09fe4f83b3)

What am I missing? Do I need to use a model other than Llama3?

**Comment by cheahjs** [comment, marked as answer]
Llama3 is not a multimodal model capable of vision input. If you're running Ollama models, you would need to use [llava](https://ollama.com/library/llava) or a finetune of it.

The poor user experience is currently due to a limitation in Ollama/OpenAI APIs not returning the capabilities of a model, and a lack of configurability on Open WebUI's end to allow admins to manually set a model's capabilities.

**Reply by gnapse** [reply]
Thanks!

**Comment by HierarchyMan** [comment]
![image](https://github.com/user-attachments/assets/a6d0644e-3e7b-43d9-8a5f-a8c84502df3a)
what about now?
still doesnt recognise do i need a new modelfile?

**Reply by delfireinoso** [reply]
start. with a model form ollama: Llava, llama3.2-vision or llava-phi3. After this you can try models form hugginface, but there you need correct modelfiles

**Reply by HierarchyMan** [reply]
Alright I will do that. Thanks

**Comment by JensSettelmeier** [comment]
Did anyone find a solution? Even for image models it does not work for me.

**Reply by rgaricano** [reply]
It have to be a config related
check that model have vision checked, that user have permisions for images, and if doesn't work look logs for errors.

**Comment by krzsam** [comment]
Checked the latest version: v0.6.30 [(latest)](https://github.com/open-webui/open-webui/releases/tag/v0.6.30)
It still does not work, and there is nothing in the logfiles (checking container logfiles via podman)
The UI looks as if the request was sent to the model, but the logfiles do not show anything, neither as request going out, nor an error from a model - so it is not clear where exactly the request was sent (if it was sent at all)

**Comment by Schnubl** [comment]
Does your chat recognize other uploaded files like .txt or .pdf or do you get the same message there?

**Reply by krzsam** [reply]
The attachment of images inside the UI works fine - they are visible in the chat window above the prompt (for example "Describe the image")
The problem is that after sending the prompt nothing really happens - the UI does look like it did send the prompt somewhere (the button in the bottom right changes to square (Stop)) - but there is no evidence in the logfile on the OpenWebUI backend side that anything happened.
Btw I am trying to use vision model through OpenAI API, not ollama
